lib/rubygems/commands/install_command.rb

```diff
@@ -260,7 +261,8 @@ def show_install_errors(errors) # :nodoc:
     errors.each do |x|
       return unless Gem::SourceFetchProblem === x
 
-      msg = "Unable to pull data from '#{x.source.uri}': #{x.error.message}"
+      printable_uri = Gem::PrintableUri.parse_uri(x.source.uri.clone)
+      msg = "Unable to pull data from '#{printable_uri}': #{x.error.message}"
```

Comment:
I've run the command by myself and I didn't see any credential inside the `x.error.message` so I kept it as it is.